vxquery-core/src/main/java/org/apache/vxquery/metadata/AbstractVXQueryDataSource.java

```diff
@@ -98,6 +100,22 @@ public void addChildSeq(int integer) {
     public List<Integer> getChildSeq() {
         return childSeq;
     }
+    
+    public void addIndexChildSeq(int integer) {
+        indexChildSeq.add(integer);
+    }
+
+    public List<Integer> getIndexChildSeq() {
+        return indexChildSeq;
+    }
+    
+    public void addIndexSeq(Byte[] index) {
```

Comment:
Should this be called addIndexValueSeq? To clarify this is for index values, not paths.